### DISCUSSION TITLE
Micronaut data with single database for multi tenant solutions

### DISCUSSION THREAD

**Discussion by maciej-sewerniak** [micronaut-projects/micronaut-data#1675] [Q&A] [answered]
Is there a possibility to dynamically apply collection name for interface in Micronaut Data for Mongo DB ReactiveStreamsCrudRepository ? 
The goal is to support multi-tenant database structure via single repository. Our collection name pattern is {{tenant}}--{{resource}} for example: tenant1--apples, tenant2--apples in apples database. 
So the target solution is to dynamically change collection name for repository based on request header value.

**Comment by graemerocher** [comment, marked as answer]
See https://github.com/micronaut-projects/micronaut-data/issues/253

**Reply by dstepanov** [reply]
Multitenancy is planned for the next release

**Reply by maciej-sewerniak** [reply]
Thanks, can't wait to join this to our system ! 👍
